### DISCUSSION TITLE
How to export ou download Table content in CSV or PDF like "bootstrap-table.com" does ???

### DISCUSSION THREAD

**Discussion by chegmarco1989** [Okipa/laravel-table#123] [Ideas]
Hello.

Seing this part circled in red for Bootstrap-table (https://examples.bootstrap-table.com/):

![image](https://user-images.githubusercontent.com/51763985/219981407-1249774f-1f0f-4eb2-944d-b1829607e346.png)

I worder if is there a way to export Table in different files (csv, pdf, etc) like it is possible with Bootstrap-table (https://examples.bootstrap-table.com/) as you can see on the part circled in green here:

![image](https://user-images.githubusercontent.com/51763985/219981532-f8733836-e8d2-44eb-8ad7-463c9edc11a5.png)

If so, how to do that ???
How to export ou download Table content in CSV or PDF exactly like "bootstrap-table.com" does ???

**Comment by Okipa** [comment]
Hi @chegmarco1989, unfortunately this is not a feature available in this package.

In fact, implementing data export in many formats as JSON, XML, CSV, TXT, SQL, Excel, etc. would need to integrate many other libraries. I don't want to add external dependencies in order to keep the maintenance as easiest as possible.

But I think this is something your definetely can implement with `laravel-table`!

I think you could set this in a Bulk Action => you would need to select the rows you want to export and trigger a `Export in CSV` for example.

Of course, you'll need to install and manipulate the third party package that would allow you to generate this export yourself.

See this part of the doc to generate your own Bulk Action 👉 https://github.com/Okipa/laravel-table#define-table-bulk-actions
